### PR TITLE
Rebuilding the GHA caches / Resolving CBC warmstart on Windows

### PR DESCRIPTION
**Pull request by jsiirola**
## Summary/Motivation:
This forces a rebuild of the GHA caches to pull a new version of the codecov.sh upload script in response to a recent Codecov security disclosure (see https://codecov.io/disclosure).

As part of the dependency rebuild, some tests started failing.  This PR also contains fixes for those failing tests.

## Changes proposed in this PR:
- rebuild the GHA download caches
- Update how `file_utils` munges relative paths so that behavior is consistent across platforms
- Fix CBC log level so that warmstart test passes with new (2.10.4) CBC

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/common/fileutils.py [hunk 1]
```diff
@@ -336,9 +336,9 @@ def find_library(libname, cwd=True, include_PATH=True, pathlist=None):
         return lib
     # Search 3: use ctypes.util.find_library (which expects 'lib' and
     # extension to be removed from the name)
-    if libname.startswith('lib') and _system() != 'windows':
-        libname = libname[3:]
     libname_base, ext = os.path.splitext(os.path.basename(libname))
+    if libname_base.startswith('lib') and _system() != 'windows':
```
Comment (mrmundt) [inline pyomo/common/fileutils.py:340]: Are these supposed to be part of this PR?

**Comment by mrmundt** [comment]
@jsiirola - What's going on with the Windows failures?

File: pyomo/common/fileutils.py (hunk 1 above)
Comment (jsiirola) [reply pyomo/common/fileutils.py:340]: Unfortunately, yes - to get tests passing again.

**Comment by jsiirola** [comment]
@mrmundt: by rebuilding the cache, we are getting a newer build of CBC, which is causing test failures.  I think I tracked it down to a change in CBC's default logging levels.  I have updated the test that depends on the more verbose logging to fix the level appropriately, so things should start passing again.

**Review by michaelbynum — APPROVED** [review]
Looks good.

**Pull request merged**